### PR TITLE
OCPBUGS-38440: [release-4.14] 4.14 subnet config

### PR DESCRIPTION
**Pull request by bpickard22**
Backport of https://github.com/openshift/cluster-network-operator/pull/2375 
Some small tweaks had to be made due to some slight code differences in 4.14 and 4.15, nothing major 

Had to hack at the hypershift deps since they are still imported in 4.14, followed @trozet steps that he took last time we had to do this

### CODE REVIEW

**Comment by bpickard22** [comment]
/hold 
waiting on qe pre merge regression tests

**Comment by asood-rh** [comment]
QE regression looks good for join switch and transit switch subnets with pre merge testing. Masq will be tested with other PR.

/label qe-approved

**Comment by asood-rh** [comment]
Verified the PR using the tests https://github.com/openshift/openshift-tests-private/blob/master/test/extended/networking/ovn_misc.go#L1165

File: pkg/network/ovn_kubernetes.go [hunk 1]
```diff
@@ -2547,5 +2519,148 @@ func annotateNodeDaemonset(kubeClient kubernetes.Interface) error {
 	}); err != nil {
 		return err
 	}
+
```
Comment (arghosh93) [inline pkg/network/ovn_kubernetes.go:2522]: Did you add this empty line intentionally? If not, we can get rid of this and not make part of this PR(nit).

File: vendor/modules.txt [hunk 2]
```diff
@@ -1614,6 +1614,7 @@ sigs.k8s.io/structured-merge-diff/v4/value
 # sigs.k8s.io/yaml v1.3.0
 ## explicit; go 1.12
 sigs.k8s.io/yaml
+# github.com/openshift/api => github.com/bpickard22/api v0.0.0-20240715224255-5ff1b2cdd555
```
Comment (arghosh93) [inline vendor/modules.txt:1617]: Need to point to api only when the open api PR gets merged to release-4.14. This comment is just for future reference.

**Comment by kyrtapz** [comment]
@bpickard22 I will leave lifting the hold to you

**Pull request merged**